### PR TITLE
Update test due to switch to intel syntax by default in #48103

### PR DESCRIPTION
**Pull request by Zentrik**
Not sure why CI wasn't failing but this fixes the test locally for me on master.

### CODE REVIEW

**Comment by Zentrik** [comment]
Actually not sure what's going on it works with the binary on juliaup.

**Comment by Zentrik** [comment]
CI looks good

File: stdlib/InteractiveUtils/test/runtests.jl [hunk 1]
```diff
@@ -547,9 +547,9 @@ if Sys.ARCH === :x86_64 || occursin(ix86, string(Sys.ARCH))
     output = replace(String(take!(buf)), r"#[^\r\n]+" => "")
     @test !occursin(rgx, output)
 
-    code_native(buf, linear_foo, ())
-    output = String(take!(buf))
-    @test occursin(rgx, output)
+    code_native(buf, linear_foo, (), debuginfo = :none)
```
Comment (giordano) [inline stdlib/InteractiveUtils/test/runtests.jl:550]: Should this also specify `syntax=:intel` not to rely on the default value?

Comment (Zentrik) [reply stdlib/InteractiveUtils/test/runtests.jl:550]: The test right before this one does that. This test seems to specifically check the default.

**Comment by giordano** [comment]
> Not sure why CI wasn't failing

Locally I get
```
julia> rgx = r"%";

julia> buf = IOBuffer();

julia> linear_foo() = 5;

julia> code_native(buf, linear_foo, ())

julia> output = String(take!(buf));

julia> print(output)
	.text
	.file	"linear_foo"
	.section	.ltext,"axl",@progbits
	.globl	julia_linear_foo_1014           # -- Begin function julia_linear_foo_1014
	.p2align	4, 0x90
	.type	julia_linear_foo_1014,@function
julia_linear_foo_1014:                  # @julia_linear_foo_1014
; Function Signature: linear_foo()
; ┌ @ REPL[3]:1 within `linear_foo`
# %bb.0:                                # %top
	push	rbp
	mov	rbp, rsp
; │ @ REPL[3] within `linear_foo`
	mov	eax, 5
	pop	rbp
	ret
.Lfunc_end0:
	.size	julia_linear_foo_1014, .Lfunc_end0-julia_linear_foo_1014
; └
                                        # -- End function
	.section	".note.GNU-stack","",@progbits
```
which technically is correct to return `true` for `occursin(rgx, output)` since there's a percentage sign in the comment, which you fixed by removing the comments and checking that there's no `%`.  Somehow you don't have any `%` sign even in the wrong version of the check?  I think your change is good, but I'm trying to understand why the botched version of the test didn't work for you.

**Comment by Zentrik** [comment]
`# %bb.0:                                # %top` is missing for me, no clue why. I am running on a fork and not sure if it replicates on master.

```
julia> linear_foo() = 5;

julia> code_native(linear_foo, ())
	.text
	.file	"linear_foo"
	.section	.ltext,"axl",@progbits
	.globl	julia_linear_foo_822
	.p2align	4
	.type	julia_linear_foo_822,@function
julia_linear_foo_822:
	push	rbp
	mov	rbp, rsp
; ┌ @ REPL[1] within `linear_foo`
	mov	eax, 5
	pop	rbp
	ret
.Lfunc_end0:
	.size	julia_linear_foo_822, .Lfunc_end0-julia_linear_foo_822

	.section	".note.GNU-stack","",@progbits
```

**Comment by giordano** [comment]
Ok, thanks, we don't know why you have different code, but at least that explains why the test failed for you.  I'm going to merge, because the change here looks an improvement in any case.

**Pull request merged**